sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -347,15 +347,16 @@ private[sql] object DataSourceScanExec {
         case _ => None
       }
 
-      def toAttribute(colName: String): Attribute = output.find(_.name == colName).getOrElse {
-        throw new AnalysisException(s"bucket column $colName not found in existing columns " +
-          s"(${output.map(_.name).mkString(", ")})")
-      }
+      def toAttribute(colName: String): Attribute = output.find(_.name == colName).get
 
       bucketSpec.map { spec =>
         val numBuckets = spec.numBuckets
-        val bucketColumns = spec.bucketColumnNames.map(toAttribute)
-        HashPartitioning(bucketColumns, numBuckets)
+        if (spec.bucketColumnNames.forall(colName => output.exists(_.name == colName))) {
+          val bucketColumns = spec.bucketColumnNames.map(toAttribute)
```

Comment:
this is OK, but I think a better way to write this is
```
val bucketColumns = output.filter { a => spec.bucketColumnNames.contains(a.name) }
if (bucketColumns.size == spec.bucketColumnNames.size) {
  HashPartitioning(bucketColumns, numBuckets)
}
```
then you don't need `toAttribute`